src/main/java/org/Lab1/Operation.java

```diff
@@ -0,0 +1,190 @@
+                    moviesList[j] = moviesList[j + 1];
+                    moviesList[j + 1] = temp;
+                }
+            }
+        }
+
+        return moviesList;
+    }
+    public static int findMovie(Scanner sc,Movie[] moviesList) {
+        int filmIndex = 0;
+        System.out.println("Enter movie Name");
+        String movieName = sc.nextLine();
+        boolean findMovie = false;
+        for (int z = 0; z < moviesList.length; z++) {
+            filmIndex = z;
+            if (Objects.equals(moviesList[z].getName(), movieName)) {
+                findMovie = true;
+                break;
+            }
+        }
+        if (findMovie) {
+
+            return filmIndex;
+
+        }else {return -1;}
+
+
+    };
+
+    public static void main(String[] args) {
+        int findResult;
+        Movie[] moviesList = new Movie[0];
+        Scanner sc = new Scanner(System.in);
+        int userSelect = 0;
+
+        while (true) {
+
+            System.out.println("Press 1 to input movies");
+            System.out.println("Press 2 to display movies and ratings");
+            System.out.println("Press 3 to input more movies.");
+            System.out.println("Press 4 to find statistics.");
+            System.out.println("Press 5 to search movie.");
+            System.out.println("Press 6 to update movie rating ");
+            System.out.println("Press 7 to delete movie.");
+            System.out.println("Press 8 to sort movie list.");
+            System.out.println("Press 0 to exit.");
+            userSelect = sc.nextInt();
+            sc.nextLine(); // Consume leftover newline
+
+            switch (userSelect) {
+                case 1:
+                    moviesList = movieAdder(moviesList, sc);
+                    break;
+
+                case 2:
+                    movieListViewer(moviesList, sc);
+                    break;
+                case 3:
+                    System.out.println("How many additional movies you want to add?:");
+                    int moiveCount = sc.nextInt();
+                    sc.nextLine();
+
+                    for (int i = 0; i < moiveCount; i++) {
+                        moviesList = movieAdder(moviesList, sc);
+                    }
+                    break;
+
+                case 4:
+                    int maxRatingIndex = 0;
+                    double maxRating = moviesList[0].getRating();
+                    double minRating = moviesList[0].getRating();
+                    int minRatingIndex = 0;
+                    double avarageRating = 0;
+                    double avarageSum = 0;
+                    for (int k = 0; k < moviesList.length; k++) {
+                        double currentRating = moviesList[k].getRating();
+                        avarageSum += currentRating;
+                        if (currentRating > maxRating) {
+                            maxRating = currentRating;
+                            maxRatingIndex = k;
```

Comment:
I think we shouldn't store indexes, at the end we have our hands maxRating and minRating) No need get these from the array again